app/test_main.py

```diff
@@ -1 +1,23 @@
-# write your code here
+from unittest import mock
+import pytest
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, has_internet, is_valid, expected",
+    [
+        ("https://www.google.com", True, True, "Accessible"),
+        ("https://www.google.com", False, True, "Not accessible"),
+        ("https://somewebsitename.com", True, False, "Not accessible"),
+        ("https://somewebsitename.com", False, False, "Not accessible")
```

Comment:
it seems to me there is no need to use different urls for this tests, so u can write an url only once directly in ur func